packages/hdwallet-native/src/crypto/isolation/adapters/ethereum.ts

```diff
@@ -9,18 +10,20 @@ function ethSigFromRecoverableSig(x: SecP256K1.RecoverableSignature): ethers.Sig
   return ethers.utils.splitSignature(core.compatibleBufferConcat([sig, Buffer.from([recoveryParam])]));
 }
 
-export class SignerAdapter extends ethers.Signer {
-  protected readonly _isolatedKey: SecP256K1.ECDSAKey;
+export class SignerAdapter {
+  protected readonly rootNode: Isolation.Adapters.BIP32;
   readonly provider?: ethers.providers.Provider;
 
-  protected constructor(isolatedKey: SecP256K1.ECDSAKey, provider?: ethers.providers.Provider) {
-    super();
-    this._isolatedKey = isolatedKey;
+  protected constructor(rootNode: Isolation.Adapters.BIP32, provider?: ethers.providers.Provider) {
+    this.rootNode = rootNode;
     this.provider = provider;
   }
 
-  static async create(isolatedKey: SecP256K1.ECDSAKey, provider?: ethers.providers.Provider): Promise<SignerAdapter> {
-    return new SignerAdapter(isolatedKey, provider);
+  static async create(
```

Comment:
we don't really need the static async create anymore if we want to clean that up